tests/fixture/e2e/env.go

```diff
@@ -144,23 +145,43 @@ func NewTestEnvironment(tc tests.TestContext, flagVars *FlagVars, desiredNetwork
 			flagVars.ReuseNetwork(),
 		)
 
-		// Wait for chains to have bootstrapped on all nodes
+		// create a slice of all the validators and chains pairs that we'll be waiting to be bootstrapped.
+		var pendingBootstappingChains []struct {
+			validatorID ids.NodeID
+			chainID     ids.ID
+		}
+		for _, subnet := range network.Subnets {
+			for _, validatorID := range subnet.ValidatorIDs {
+				for _, chain := range subnet.Chains {
+					pendingBootstappingChains = append(pendingBootstappingChains, struct {
+						validatorID ids.NodeID
+						chainID     ids.ID
+					}{validatorID: validatorID, chainID: chain.ChainID})
+				}
+			}
+		}
+		// Wait for chains to have bootstrapped on all validators
 		tc.Eventually(func() bool {
-			for _, subnet := range network.Subnets {
-				for _, validatorID := range subnet.ValidatorIDs {
-					uri, err := network.GetURIForNodeID(validatorID)
-					require.NoError(err)
-					infoClient := info.NewClient(uri)
-					for _, chain := range subnet.Chains {
-						isBootstrapped, err := infoClient.IsBootstrapped(tc.DefaultContext(), chain.ChainID.String())
-						// Ignore errors since a chain id that is not yet known will result in a recoverable error.
-						if err != nil || !isBootstrapped {
-							return false
-						}
-					}
+			allValidatorsBootstrapped := true
+			for i := len(pendingBootstappingChains) - 1; i >= 0; i-- {
+				pending := pendingBootstappingChains[i]
+				uri, err := network.GetURIForNodeID(pending.validatorID)
+				require.NoError(err)
+				infoClient := info.NewClient(uri)
+				isBootstrapped, err := infoClient.IsBootstrapped(tc.DefaultContext(), pending.chainID.String())
+				// Ignore errors since a chain id that is not yet known will result in a recoverable error.
+				if err != nil || !isBootstrapped {
+					allValidatorsBootstrapped = false
+					continue
```

Comment:
Why do we perform an `IsBootstrapped` check here rather than checking if the node is healthy? If the node reports healthy, all bootstrapping checks should pass and this will include correctly checking that dynamic state sync has completed (which is handled as a bit of an outlier).

Reply:
by the time we reach this point, the health check was already complete against all the node. Some differences:
- The health checking doesn't seem to be chain-specific, whereas the bootstrapping test is.
- When doing the health checking, the port would get dynamically refreshed. That's won't happen during the bootstrapping testing.